File: backend/services/subscription/src/main/java/pl/muybien/subscription/data/SubscriptionDetail.java
package pl.muybien.subscription.data;

import pl.muybien.subscription.AssetType;
import pl.muybien.subscription.CurrencyType;
import pl.muybien.subscription.NotificationType;

import java.time.LocalDateTime;

public record SubscriptionDetail(
        String id,
        String uri,
        String customerId,
        String target,
        String financeName,
        CurrencyType requestedCurrency,
        Double upperBoundPrice,
        Double lowerBoundPrice,
        AssetType assetType,
        NotificationType notificationType,
        LocalDateTime createdDate
) {
}

File: frontend/src/app/common/model/subscription-detail.ts
export class SubscriptionDetail {
    constructor(
        public id: string,
        public financeName: string,
        public uri: string,
        public upperBoundPrice: number,
        public lowerBoundPrice: number,
        public assetType: string,
        public notificationType: string,
        public createdDate: string
    ) {
    }
}
